### PR TITLE
odbc: fix cross-build to iOS + modernize

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/odbc/all/conanfile.py [hunk 1]
```diff
@@ -65,9 +71,17 @@ def _configure_autotools(self):
         self._autotools.configure(configure_dir=self._source_subfolder, args=args)
         return self._autotools
 
+    @property
+    def _user_info_build(self):
+        return getattr(self, "user_info_build", None) or self.deps_user_info
```
Comment (madebr) [inline recipes/odbc/all/conanfile.py:76]:
```suggestion
        return getattr(self, "user_info_build", self.deps_user_info)
```

**Review by madebr — DISMISSED** [review]

File: recipes/odbc/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/odbc/all/conanfile.py]: Could also be used for _settings_build in others recipe:

```python
    @property
    def _settings_build(self):
        return getattr(self, "settings_build", self.settings)
```

File: recipes/odbc/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -2,7 +2,7 @@ cmake_minimum_required(VERSION 3.1)
 project(test_package C)
 
 include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-conan_basic_setup()
+conan_basic_setup(TARGETS)
```
Comment (ericriff) [inline recipes/odbc/all/test_package/CMakeLists.txt:5]: Quick question if you don't mind.
Why is this change needed? Wouldn't the following `find_package()` set the targets?

**Review by ericriff — APPROVED** [review]

File: recipes/odbc/all/test_package/CMakeLists.txt (hunk 2 above)
Comment (SpaceIm) [reply recipes/odbc/all/test_package/CMakeLists.txt:5]: It's orthogonal. The purpose here is to not include headers folders through conan_basic_setup().

**Review by SSE4 — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]
